### PR TITLE
Add kuttl support for openstack-operator

### PR DESCRIPTION
**Pull request by jparoly**
Depends on: https://github.com/openstack-k8s-operators/openstack-operator/pull/323

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1227,6 +1230,21 @@ horizon_kuttl: namespace input openstack_crds openstack_storage_crds deploy_clea
...
+.PHONY: openstack_kuttl
+openstack_kuttl: namespace input openstack_crds deploy_cleanup mariadb mariadb_deploy keystone rabbitmq keystone_deploy rabbitmq_deploy openstack openstack_deploy ## runs kuttl tests for the openstack operator. Installs openstack crds and mariadb, keystone, rabbitmq, ovn, infra and openstack operators and cleans up previous deployments before running the tests and, add cleanup after running the tests.
+	$(eval $(call vars,$@,openstack))
+	make wait
+	make openstack_kuttl_run
+	make rabbitmq_deploy_cleanup
+	make deploy_cleanup
+	make openstack_cleanup
+	make keystone_cleanup
+	make mariadb_cleanup
```
Comment (abays) [inline Makefile:1246]: RabbitMQ, MariaDB and Keystone operators are all installed by the OpenStack operator itself.  So they should not be listed here as dependency targets nor should we call their cleanup targets.

Comment (jparoly) [reply Makefile]: Thank you, updated.

File: Makefile [hunk 2]
```diff
@@ -1265,6 +1268,18 @@ horizon_kuttl: namespace input openstack_crds openstack_storage_crds deploy_clea
 	make cleanup
 	make infra_cleanup
 
+.PHONY: openstack_kuttl_run
+openstack_kuttl_run: ## runs kuttl tests for the openstack operator, assumes that everything needed for running the test was deployed beforehand.
+	kubectl-kuttl test --config ${OPENSTACK_KUTTL_CONF} ${OPENSTACK_KUTTL_DIR}
+
+.PHONY: openstack_kuttl
+openstack_kuttl: namespace input openstack_crds deploy_cleanup openstack openstack_deploy ## runs kuttl tests for the openstack operator. Installs openstack crds and the openstack operator, cleans up previous deployments before running the tests and, cleans up after running the tests.
```
Comment (cescgina) [inline Makefile:1276]: In this case I think we don't need `openstack_crds` as `openstack` already install the crds. Also, we probably don't need to run `openstack_deploy` as a dependency if the tests themselves apply a sample

File: Makefile [hunk 3]
```diff
@@ -1265,6 +1268,18 @@ horizon_kuttl: namespace input openstack_crds openstack_storage_crds deploy_clea
...
+	$(eval $(call vars,$@,openstack))
+	make wait
+	make openstack_kuttl_run
+	make deploy_cleanup
```
Comment (cescgina) [inline Makefile:1280]: `deploy_cleanup` should be `openstack_deploy_cleanup`

File: Makefile (hunk 2 above)
Comment (jparoly) [reply Makefile:1276]: good points, thank you

**Comment by abays** [comment]
We will also need something similar to this [1] and this [2].

[1] https://github.com/openstack-k8s-operators/install_yamls/blob/99711808f7efd2db30814c93f1ac46d99c054594/Makefile#L64
[2] https://github.com/openstack-k8s-operators/install_yamls/blob/99711808f7efd2db30814c93f1ac46d99c054594/Makefile#LL1032C1-L1033C77

**Comment by jparoly** [comment]
errors are: `public endpoint for network service in regionOne region not found
command terminated with exit code 1`  

known?

**Comment by karelyatin** [comment]
> errors are: `public endpoint for network service in regionOne region not found command terminated with exit code 1`
> 
> known?

yes recently fixed with https://github.com/openstack-k8s-operators/openstack-operator/pull/328

**Comment by abays** [comment]
recheck

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**